### ISSUE TITLE
Add servo_actuator peripheral class with sense_pin, purpose, and trigger schedule

### ISSUE THREAD

**Issue by renanmzmendes** [fishhub-oss/fishhub-firmware#91]
Implements the `servo_actuator` peripheral kind. This is the first FishHub-native actuator peripheral, introduced alongside the automatic fish feeder but designed generically — the class does not know it is a feeder.

#### Files

- `src/peripherals/servo_actuator.h`
- `src/peripherals/servo_actuator.cpp`
- Update `peripheral_manager.cpp` — add `servo_actuator` to kind-dispatch
- Update `mqtt_client.cpp` — add `servo_actuator` to `onPeripheralConfig()` kind-dispatch
- Update `include/pins.h` — add `SERVO_SENSE_PIN 18`, update `SERVO_PIN` comment
- Update `CLAUDE.md` — pin table

**PlatformIO dependency:**

```
lib_deps =
  ; ... existing ...
  madhephaestus/ESP32Servo @ ^3.0.5
```

---

#### `purpose` field

A new optional string field stored in the peripheral NVS config alongside `kind`, `pin`, etc. The firmware stores and echoes `purpose` but does not interpret it. The server and UI switch on `purpose` to apply human-friendly labels and actions.

---

#### Actuation behaviour

On receiving an `actuate` command or when a trigger schedule fires:

1. `servo.attach(pin)`
2. `servo.writeMicroseconds(2000)` — full CW rotation
3. `delay(rotation_ms)`
4. `servo.writeMicroseconds(1500)` — stop
5. `servo.detach()` — release PWM; servo goes limp (no heat buildup)

---

#### `sense_pin` (optional)

When `sense_pin` is configured (not `-1`):

- `pinMode(sense_pin, INPUT_PULLUP)` in `begin()`
- Each `tick()`: invert `digitalRead(sense_pin)` → `sense` value (1 = beam interrupted, 0 = beam clear)
- Append `<name>/sense` SenML record to next publish batch

When `sense_pin` is absent, peripheral operates as a plain servo actuator with no feedback.

---

#### Trigger schedule

`servo_actuator` uses a **trigger-based schedule**, distinct from the relay's window-based schedule. A trigger fires exactly once per cron occurrence. `last_fired` per trigger id is persisted to NVS to guarantee idempotency across reboots.

NVS config shape:

```json
{
  "triggers": [
    { "id": "a3f1", "cron": "0 8 * * *",  "value": 1500 },
    { "id": "b2e7", "cron": "0 20 * * *", "value": 1500 }
  ],
  "last_fired": {
    "a3f1": 1746000000,
    "b2e7": 1745913600
  }
}
```

`tick()` logic — edge detection, not level:

```cpp
for (auto& trigger : _schedule.triggers) {
  if (trigger.isDue(now) &&
      trigger.lastFired < startOfCurrentOccurrence(now, trigger.cron)) {
    _actuate(trigger.value);
    trigger.lastFired = now;
    _persistLastFired();  // write back to NVS
  }
}
```

---

#### MQTT command payload

Topic: `fishhub/<device_id>/peripherals/<name>/cmd`

```json
{ "op": "actuate", "rotation_ms": 1500 }
```

#### SenML output

```json
[
  { "bn": "fishhub/device/", "bt": 1746000000 },
  { "n": "feeder0/rotation", "u": "ms", "v": 1500 },
  { "n": "feeder0/sense",    "u": "/",  "v": 1    }
]
```

`rotation` published only after actuation. `sense` published every tick when `sense_pin` is configured.

#### Peripheral registration payload

```json
{ "op": "create", "kind": "servo_actuator", "purpose": "feeder", "pin": 17, "sense_pin": 18 }
```

---

#### Acceptance criteria

- [ ] `servo_actuator` can be registered via MQTT `create` with and without `sense_pin` and `purpose`
- [ ] `actuate` command spins the servo for the specified duration and stops cleanly
- [ ] Servo is detached after each actuation
- [ ] `rotation` SenML record published after each actuation
- [ ] `sense` SenML record published every tick when `sense_pin` is configured
- [ ] `sense = 1` when break-beam interrupted, `0` when clear
- [ ] Trigger schedule fires each trigger exactly once per cron occurrence
- [ ] `last_fired` is persisted to NVS; no re-fire after reboot for an already-executed trigger
- [ ] `purpose` is stored in NVS and echoed in status payloads
- [ ] `pio run` compiles without warnings
